service/history/workflow/update/update.go

```diff
@@ -494,6 +502,34 @@ func (u *Update) onResponseMsg(
 	return nil
 }
 
+// isIncomplete checks if update is already completed (rejected or processed).
+func (u *Update) isIncomplete() bool {
+	return !u.state.Matches(stateSet(stateProvisionallyCompleted | stateCompleted))
+}
+
+// CancelIncomplete cancels update if it wasn't completed yet:
+//   - if in stateAdmitted, stateRequested, or stateSent -> reject,
+//   - if in stateAccepted -> do nothing,
+//   - if in stateCompleted -> do nothing.
+func (u *Update) CancelIncomplete(ctx context.Context, reason CancelReason, eventStore EventStore) error {
+	if u.state.Matches(stateSet(stateAdmitted | stateProvisionallyRequested | stateRequested | stateProvisionallySent | stateSent)) {
+		return u.reject(ctx, reason.RejectionFailure(), eventStore)
```

Comment:
What I love about comments, is that they frequently lye. Docstring for this function does not match the functionality.

Reply:
Do you mean that I didn't includes Provisional states in docstring? This is on purpose. It is not not important detail and I want docstring to explain what this method does, not to rewrite in plain English.